driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -115,13 +123,17 @@ public InternalConnection get(final long timeout, final TimeUnit timeUnit) {
                         pooledConnection.open();
                     } catch (Throwable t) {
                         pool.release(pooledConnection.wrapped, true);
+                        connectionPoolListener.connectionCheckOutFailed(new ConnectionCheckOutFailedEvent(serverId,
+                                Reason.CONNECTION_ERROR));
                         if (t instanceof MongoException) {
                             throw (MongoException) t;
                         } else {
                             throw new MongoInternalException(t.toString(), t);
                         }
                     }
                 }
+                connectionPoolListener.connectionCheckedOut(
+                        new ConnectionCheckedOutEvent(pooledConnection.getDescription().getConnectionId()));
 
                 return pooledConnection;
             } finally {
```

Comment:
Seems like there would need to be a catch clause here as well to handle any other failures to check out a connection (e.g a timeout).  In those cases we would also need a checkout failed event.  You'll have to be careful how to track the Reason properly, given all the different failure scenarios.

Reply:
I reviewed it again, and I agree with you.